Update documentation about `@abstract` method annotation in RBS (#8908)
sorbet/sorbet: website/docs/rbs-support.md

The following signatures are equivalent:

```ruby
# @abstract
#: (Integer) -> void
def foo1(x); end

sig { abstract.params(x: Integer).void }
def foo2(x); end

> 

Reviewer: wdyt keeping this example with another annotation like `@override`?

Author: There is one just below ⬇️ Or let me know if you were thinking about something else

Reviewer: I'm blind 🤦

## Unsupported features

### Abstract methods

While the `@abstract` annotation is technically supported for methods, it is
currently not recommended. To understand why, take the following example:

```ruby
# @abstract
class Foo
  # @abstract

>   # () -> void

Reviewer: ```suggestion
  # -> void
```

  def foo; end
end

class Bar < Foo
  # @override

>   # () -> void

Reviewer: ```suggestion
  # -> void
```

  def foo
    super
  end
end
```

When using traditional Sorbet sigs, the call to `super` inside of `Bar#foo`
would error at runtime, because `sorbet-runtime` uses metaprogramming to
remove the `Foo#foo` method. Any mistaken calls to this method would raise an
error at runtime.

However, RBS signatures have no runtime component. This means that any
inadvertent calls to abstract methods will silently no-op, which has the

> potential to create subtle bugs and unintended behavior in our applications.

Reviewer: ```suggestion
potential to create subtle bugs and unintended behavior in production.
```

Instead of using the `@abstract` annotation on methods, place any abstract
methods in a separate `.rbi` shim file.

Modifying the example above, place only the abstract class definition in a
Ruby file:

```ruby

> # @abstract

Reviewer: Honestly at this point they could also move the class? 🤔

Author: Oh let me clarify the example and include Bar.

Author: Is the current example clearer? My intention was to keep the definition of `Foo` in the Ruby files because it has to exist at runtime.
